### ISSUE TITLE
Returned value should be not_found() if secret not found

### ISSUE THREAD

**Issue by petems** [petems/petems-hiera_vault#10]
If I do a Vault lookup for a secret, and it can't be found, currently the code returns nil, instead of returning the `not_found` method.

@hlindberg explained it well on the Puppet slack:

```
hlindbe [13:56]
yep - it finds a `nil`
the hiera-vault function should not return `nil` if it detects a “not found” - it should call the context method `not_found` instead
if it returns a `nil` it means that the value IS `undef` in puppet - i.e. a valid value
```

This means, with Puppet code like this:

```puppet
$vault_notify = lookup({"name" => "vault_notify", "default_value" => "No Vault Secret Found"})
notify { "testing vault ${vault_notify}":}
```

It will return a string with an undef value, rather than seeing that it wasnt found and returning the default value:

```
Notice: /Stage[main]/Profile::Vault_message/Notify[testing vault ]/message: defined 'message' as 'testing vault '
```

**Comment by petems** [comment]
A quick test with the following change appears to work:

```diff
+      context.not_found if secret.nil?
      next if secret.nil?
```

**Comment by petems** [comment]
Fixed in 105b37f428463a6f030f27b06798d38c773484e5 from https://github.com/petems/petems-hiera_vault/pull/7
